custom-carbon-log-appender/README.md

```diff
@@ -0,0 +1,41 @@
+# Custom Carbon Log Appender.
+## Introduction / Use case
+This project is aimed to implement a simple log appender, that can change the default logs of WSO2 Identity Server. The exact use case is as follows.
+
+If we have log entries as follows in our code base, with CRLF (**C**arriage **R**eturn (ASCII 13, \r) **L**ine **F**eed (ASCII 10, \n))characters, the log file can be get disrupted.
+
+- Code.
+```
+log.info("TEST LOG =============");
+log.info("TEST LOG \n some string \r some other string \n John Doe");
+log.info("TEST LOG =============");
+```
+- Sample output.
+```
+[2018-09-04 20:39:00,661]  INFO {org.wso2.carbon.user.core.ldap.ReadOnlyLDAPUserStoreManager} -  TEST LOG =============
+[2018-09-04 20:39:00,663]  INFO {org.wso2.carbon.user.core.ldap.ReadOnlyLDAPUserStoreManager} -  TEST LOG 
+ some string
+ some other string 
+ John Doe
+[2018-09-04 20:39:00,663]  INFO {org.wso2.carbon.user.core.ldap.ReadOnlyLDAPUserStoreManager} -  TEST LOG =============
+```
+This is a known security vulnerability known as [CRLF Injection](https://www.owasp.org/index.php/CRLF_Injection). Our solution is to add a log appender which replaces above mentioned CRLF characters with "**\_**" and add a specific keyword (*Sanitized*) so that we can search and identify such entries in the future.
+
+## Applicable product version(s)
+Originlly desinged for WSO2 Identity Server version 5.1.0
+
+## How to use
+1. Build the project.
+2. Copy the JAR file in ```target``` directory into ```<IS_HOME>/repository/components/dropins``` directory.
+3. Open the file *<IS_HOME>/repository/conf/log4j.properties* file in a text editor and change,
+  ```log4j.appender.CARBON_LOGFILE=org.wso2.carbon.utils.logging.appenders.CarbonDailyRollingFileAppender```
+  to
+  ```log4j.appender.CARBON_LOGFILE=org.wso2.carbon.custom.utils.logging.appenders.CarbonDailyRollingSanitizedFileAppender```
+4. Restart the server. 
```

Comment:
What are the action we should do on Identity server to print those logs?

Reply:
I added a patch with logs. Should we add that jar to the repo as well?